lib/msal-node/test/client/ManagedIdentityApplication.spec.ts

```diff
@@ -194,6 +199,103 @@ describe("ManagedIdentityApplication unit tests", () => {
         });
     });
 
+    describe("Acquires a token successfully via an Azure Arc Managed Identity", () => {
```

Comment:
Add tests in all the sources to verify general error scenarios:
1. Response is empty
2. Response has status code 200 but the response body is null
3. Error message is parsed when the scopes are wrong

Another test case is to acquire a token for scope 1 and acquire a token for scope 2 and both should be acquired from IDP and not cache.

Reply:
I'm going to make this a separate PR since it involves all sources.